Adds 13th batch of doc text (#42407)
openshift/openshift-docs: release_notes/ocp-4-10-release-notes.adoc

[id="ocp-4-10-networking-bug-fixes"]
==== Networking

> * Previously, when Kuryr was used in a restricted installation with Proxy, the Cluster Networking Operator was not enforcing usage of Proxy to allow communication the {rh-openstack-first} API. Consequently, cluster installation did not progress. With this update, the Cluster Network Operator can communicate with the {rh-openstack} API through the Proxy. As a result, installation now succeeds. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1985486[*BZ#1985486*])

Reviewer: ```suggestion
* Previously, when Kuryr was used in a restricted installation with a proxy, the Cluster Networking Operator was not enforcing usage of the proxy to allow communication with the {rh-openstack-first} API. Consequently, cluster installation did not progress. With this update, the Cluster Network Operator can communicate with the {rh-openstack} API through the proxy. As a result, installation now succeeds. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1985486[*BZ#1985486*])
```

[id="ocp-4-10-routing-bug-fixes"]
==== Routing

* Previously, if the cluster administrator provided a default ingress certificate that was missing the newline character for the last line, the {product-title} router would write out a corrupt PEM file for HAProxy. Now, it writes out a valid PEM file even if the input is missing a newline character. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1894431[*BZ#1894431*])

> * Previously, a route created where the combined name and namespace for the DNS segment greater than 63 characters long would be rejected. This expected behavior could cause problems integrating with upgraded versions of {product-title}. Now, an annotation allows non-conformant DNS hostnames. With `AllowNonDNSCompliantHostAnnotation` set to `true`, the non-conformant DNS hostname, or one longer than 63 characters, is allowed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1964112[*BZ#1964112*])

Reviewer: ```suggestion
* Previously, a route created where the combined name and namespace for the DNS segment was greater than 63 characters long would be rejected. This behavior could cause problems integrating with upgraded versions of {product-title}. Now, an annotation allows non-conformant DNS hostnames. With `AllowNonDNSCompliantHostAnnotation` set to `true`, the non-conformant DNS hostname, or one longer than 63 characters, is allowed. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1964112[*BZ#1964112*])
```

> * Prevoiusly, the Cluster Ingress Operator would not create wildcard DNS records for Ingress Controllers when the cluster's `ControlPlaneTopology` was set to `External`. In Hypershift clusters where the `ControlPlaneTopology` was set to `External` and the Platform is AWS, the Cluster Ingress Operator never became available. This updates limits the disabling of DNS updates when the `ControlPlaneTopology` is `External` and the platform is IBMCloud. As a result, wildcard DNS entries get created for Hypershift clusters running on AWS. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011972[*BZ#2011972*])

Reviewer: ```suggestion
* Prevoiusly, the Cluster Ingress Operator would not create wildcard DNS records for Ingress Controllers when the cluster's `ControlPlaneTopology` was set to `External`. In Hypershift clusters where the `ControlPlaneTopology` was set to `External` and the platform was AWS, the Cluster Ingress Operator never became available. This updates limits the disabling of DNS updates when the `ControlPlaneTopology` is `External` and the platform is IBM Cloud. As a result, wildcard DNS entries are created for Hypershift clusters running on AWS. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2011972[*BZ#2011972*])
```

> * Previously, the cluster ingress router was blocked from working because the Ingress Operator failed to configure a wildcard DNS record for the cluster ingress router on Azure Stack Hub IPI. With this fix, the Ingress Operator now uses the configured ARM endpoint to configure DNS on Azure Stack Hub IPI. As a result, the cluster ingress router now wrks properly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2032566[*BZ#2032566*])

Reviewer: ```suggestion
* Previously, the cluster ingress router was blocked from working because the Ingress Operator failed to configure a wildcard DNS record for the cluster ingress router on Azure Stack Hub IPI. With this fix, the Ingress Operator now uses the configured ARM endpoint to configure DNS on Azure Stack Hub IPI. As a result, the cluster ingress router now works properly. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2032566[*BZ#2032566*])
```

* The Special Resource Operator (SRO) might fail to install on Google Cloud Platform due to a software-defined network policy. As a result, the simple-kmod pod is not created. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1996916[*BZ#1996916*])

> * Currently, idling a StatefulSet is unsupported when you run `oc idle` for a service that is mapped to a StatefulSet. There is no known workaround at this time. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1976894[*BZ#1976894*])

Reviewer: ```suggestion
* Currently, idling a stateful set is unsupported when you run `oc idle` for a service that is mapped to a stateful set. There is no known workaround at this time. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1976894[*BZ#1976894*])
```
